src/sdk/cpp/lib/src/iotea.cpp

```diff
@@ -321,32 +359,19 @@ void CallContext::Reply(const json& value) const {
 Gatherer::Gatherer(timeout_func_ptr timeout_func, const std::vector<CallToken>& tokens, int64_t now_ms)
     : timeout_func_{timeout_func} {
 
-    auto smallest_timeout = int64_t{-1};
+    auto nearest_timeout = std::numeric_limits<int64_t>::max();
 
     for (const auto& t : tokens) {
         ids_.insert(t.GetCallId());
 
-        auto token_timeout = t.GetTimeout();
-        if (token_timeout > 0) {
-            smallest_timeout = smallest_timeout < 0 ? token_timeout : std::min(smallest_timeout, token_timeout);
-        }
+        nearest_timeout = std::min(nearest_timeout, t.GetTimeout());
     }
 
-    if (smallest_timeout <= 0) {
-        // No timeout was set
-        timeout_ = 0;
-        return;
-    }
-
-    timeout_ = now_ms + smallest_timeout;
+    timeout_ = now_ms + nearest_timeout;
 }
 
-bool Gatherer::HasTimedOut(int64_t now) const {
-    if (timeout_ <= 0 || now < timeout_) {
-        return false;
-    }
-
-    return true;
+bool Gatherer::HasTimedOut(int64_t now_ms) const {
+    return timeout_ <= now_ms;
```

Comment:
Previously, there was the check for a timeout <= 0 --> I assume that was used to remove any timeout but I like to that a valid positive timeout has to be set (to avoid unknown behavior). With this change, is it still intended to use a signed integer or can it be replaced with an unsigned integer (uint32_t?) as an optimization?

Reply:
Before this change we accepted negative timeouts as way of saying "never timeout". This has now been changed and only finite timeouts are accepted. The timeout could be given as an `uint64_t` (this is epoch time in ms so 32-bits isn't enough) but all times in the SDK are now (and unfortunately not type aliased) int64_t so I'd like to keep it that way in order to avoid comparing different types.

Reply:
👍